mtags/src/main/scala-2/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -19,33 +18,40 @@ import org.eclipse.lsp4j.SemanticTokenTypes
  */
 final class SemanticTokenProvider(
     protected val cp: MetalsGlobal, // compiler
-    val params: VirtualFileParams
+    val params: OffsetParams
 ) {
+  import cp._
+
   val capableTypes = TokenTypes
   val capableModifiers = TokenModifiers
-
   // Alias for long notation
   val getTypeId: Map[String, Int] = capableTypes.zipWithIndex.toMap
   val getModifierId: Map[String, Int] = capableModifiers.zipWithIndex.toMap
 
-  // Initialize Tree
-  import cp._
-  val unit: RichCompilationUnit = cp.addCompilationUnit(
-    params.text(),
-    params.uri().toString(),
-    None
-  )
-  cp.typeCheck(unit) // initializing unit
-  val (root, source) = (unit.lastBody, unit.source)
   implicit val ord: Ordering[NodeInfo] = Ordering.fromLessThan((ni1, ni2) =>
     if (ni1.pos.start == ni2.pos.start) ni1.pos.end < ni2.pos.end
     else ni1.pos.start < ni2.pos.start
   )
 
-  def unitPos(offset: Int): Position = unit.position(offset)
-  val nodes: List[NodeInfo] = traverser
-    .traverse(SortedSet.empty[NodeInfo], root)
-    .toList
+  // Initialize Tree
+  case class NodeInfo(
+      sym: Option[Collector.compiler.Symbol],
```

Comment:
Should it always contain symbol? Do we use it when it's none?

Reply:
You're right. For now, there is None if symbol would be NoSymbol, but we can just keep the NoSymbol and skip the Option.